x-pack/plugins/security/common/model/kibana_privileges/feature_privileges.ts

```diff
@@ -19,7 +19,12 @@ export class KibanaFeaturePrivileges {
   }
 
   public getPrivileges(featureId: string): string[] {
-    return Object.keys(this.featurePrivilegesMap[featureId]);
+    const featurePrivileges = this.featurePrivilegesMap[featureId];
+    if (featurePrivileges == null) {
```

Comment:
thoughts on either `if (!featurePrivileges)` or `if (featurePrivileges === null)`? Here and elsewhere in this PR, I noticed a lot of loose equality checks (`==` as opposed to `===`). I know we've tried to limit using those in the past - is that still something we want to do?

Reply:
In this situation, the `RawKibanaFeaturePrivileges` interface is the following:

```
interface RawKibanaFeaturePrivileges {
  [featureId: string]: {
    [privilegeId: string]: string[];
  };
}
```

so the two following are valid
```
const withUndefined: RawKibanaFeaturePrivileges = {
  something: undefined
};

const withNull: RawKibanaFeaturePrivileges = {
  something: null
};
```

I was using `== null` here to effectively check for either `null` or `undefined`, and if that's the situation return an empty array. IIRC, both ML and Monitoring are features and there's a code-path which tries to get the privileges for these features, but there are none, since they're using the special "reserved bag".

The other places which use `== null` should be in a similar situation where technically based on the types we could have either `null` or `undefined`. Doing the whole `if (!featurePrivileges)` felt wrong since it relies on the whole truthy/falsy values of JS.